lib/ansible/plugins/test/core.py

```diff
@@ -149,7 +149,10 @@ def regex(value='', pattern='', ignorecase=False, multiline=False, match_type='s
     if multiline:
         flags |= re.M
     _re = re.compile(pattern, flags=flags)
-    return bool(getattr(_re, match_type, 'search')(value))
+    try:
+        return bool(getattr(_re, match_type)(value))
+    except (TypeError, AttributeError) as exc:
+        raise errors.AnsibleTemplatePluginError("The regex test expects a valid match_type such as 'search' or 'match'.", obj=match_type) from exc
```

Comment:
I don't understand if the obj is useful here. It appends a bit onto the error like:

```console
<<< caused by >>>

The regex test expects a valid match_type such as 'search' or 'match': 're.Pattern' object has no attribute 'missing'
Origin: <unknown>

missing
```

If I remove just the obj parameter
```suggestion
        raise errors.AnsibleTemplatePluginError("The regex test expects a valid match_type such as 'search' or 'match'.") from exc
```

I see a slightly more concise error message containing all the same information.

Reply:
The error message adapts based on what `obj` is. If it's an inline string you'll see the `Origin: <unknown>`. However, if it was from a variable, you'll see where that variable was sourced from.

Reply:
Ahh, that makes sense, thanks for the explanation. I was expecting an arrow pointing towards the inline string.